tests/telemetry/test_writer.py

```diff
@@ -637,3 +643,21 @@ def test_telemetry_writer_agentless_disabled_without_api_key():
         assert new_telemetry_writer._client._endpoint == "api/v2/apmtelemetry"
         assert new_telemetry_writer._client._telemetry_url == "https://all-http-intake.logs.datad0g.com"
         assert "dd-api-key" not in new_telemetry_writer._client._headers
+
+
+def test_telemetry_writer_is_using_agentless_by_default_if_api_key_is_available():
+    with override_global_config({"_dd_site": "datad0g.com", "_dd_api_key": "foobarkey"}):
+        new_telemetry_writer = ddtrace.internal.telemetry.TelemetryWriter()
+        assert new_telemetry_writer._enabled
+        assert new_telemetry_writer._client._endpoint == "api/v2/apmtelemetry"
+        assert new_telemetry_writer._client._telemetry_url == "https://all-http-intake.logs.datad0g.com"
+        assert new_telemetry_writer._client._headers["dd-api-key"] == "foobarkey"
+
+
+def test_telemetry_writer_is_using_agent_by_default_if_api_key_is_not_available():
+    with override_global_config({"_dd_api_key": None, "_ci_visibility_agentless_enabled": False}):
+        new_telemetry_writer = ddtrace.internal.telemetry.TelemetryWriter()
+        assert new_telemetry_writer._enabled
+        assert new_telemetry_writer._client._endpoint == "telemetry/proxy/api/v2/apmtelemetry"
+        assert new_telemetry_writer._client._telemetry_url == "http://localhost:9126"
```

Comment:
This will fix the failing test in ci but make it fail locally. In gitlab we run the testagent with the name: testagent
```suggestion
        assert new_telemetry_writer._client._telemetry_url == "http://testagent:9126"
```

Reply:
Can we check both cases instead of only passing for CI or local?
```python
assert new_telemetry_writer._client._telemetry_url in ("http://localhost:9126", "http://testagent:9126")
```